api/edge_api/identities/serializers.py

```diff
@@ -0,0 +1,127 @@
+from flag_engine.features.models import (
+    FeatureStateModel as EngineFeatureStateModel,
+)
+from flag_engine.features.models import (
+    MultivariateFeatureStateValueModel as EngineMultivariateFeatureStateValueModel,
+)
+from flag_engine.features.schemas import MultivariateFeatureStateValueSchema
+from flag_engine.identities.builders import build_identity_dict
+from flag_engine.utils.exceptions import (
+    DuplicateFeatureState,
+    InvalidPercentageAllocation,
+)
+from rest_framework import serializers
+
+from environments.identities.models import Identity
+from features.models import Feature, FeatureState, FeatureStateValue
+from features.multivariate.models import (
+    MultivariateFeatureOption,
+    MultivariateFeatureStateValue,
+)
+from features.serializers import FeatureStateValueSerializer
+
+engine_multi_fs_value_schema = MultivariateFeatureStateValueSchema()
+
+
+class EdgeMultivariateFeatureOptionField(serializers.Field):
+    def to_internal_value(self, data):
+        return MultivariateFeatureOption.objects.get(id=data)
+
+    def to_representation(self, obj):
+        return obj.id
+
+
+class EdgeMultivariateFeatureStateValueSerializer(serializers.ModelSerializer):
+    multivariate_feature_option = EdgeMultivariateFeatureOptionField()
+
+    def to_internal_value(self, data):
+        data = super().to_internal_value(data)
+        return EngineMultivariateFeatureStateValueModel(**data)
+
+    class Meta:
+        model = MultivariateFeatureStateValue
+        fields = (
+            "multivariate_feature_option",
+            "percentage_allocation",
+        )
+
+
+class FeatureStateValueEdgeIdentityField(serializers.Field):
+    def to_representation(self, obj):
+        identity_id = self.parent.get_identity_uuid()
+        return obj.get_value(identity_id=identity_id)
+
+    def get_attribute(self, instance):
+        # We pass the object instance onto `to_representation`,
+        # not just the field attribute.
+        return instance
```

Comment:
Thinking about it, doesn't it make more sense to have the logic above here instead? So we can remove the overridden `to_representation` method and instead bring that logic into `get_attribute` ? That's kind of what we're doing, right?

Reply:
That sort of goes against the design: https://www.django-rest-framework.org/api-guide/fields/#a-basic-custom-field, right?

Reply:
Weird. It definitely feels hackier to me but if the DRF docs are suggesting to do it that way then maybe it's all good.